### DISCUSSION TITLE
Why do the docs fill a form with playwright before starting the relevant step?

### DISCUSSION THREAD

**Discussion by dyllandry** [artilleryio/artillery#2399] [Artillery]
Looking at the docs for the [playwright engine `test.step`](https://www.artillery.io/docs/reference/engines/playwright#teststep-argument) function, why do the username and password fields get filled out before the `submit_login` step begins? My intuition was that filling the fields should be recorded as part of the step.

Relevant code from page:

> The final argument of the function is test, which contains the step property. The API for test.step is similar to Playwright's own test.step, which allows you to re-use similar code. The purpose in Artillery is slightly different: to emit custom metrics that represent how long each step takes.

```javascript
async function loginSearchAndLogout(page, vuContext, events, test) {
   //1. simply add this line to your scenario function, or use test.step below instead
  const { step } = test;
  const userid = vuContext.vars.userid;
  const recordid = vuContext.vars.recordid;
 
  //2. wrap any logic you have in steps (sometimes you might already have something like this done from existing playwright tests)
  await step("landing_page", async () => {
    await page.goto('https://internaltesturl.com/landing');
    await page.getByLabel('id-label').fill( `${userid}` );
    await page.getByLabel('Password').fill(`${password}`);
  })
 
  await step("submit_login", async () => {
    await page.getByRole('button', { name: 'Submit' }).click();
    await page.getByPlaceholder('enter request id').fill(`${recordid}`);
  })
 
  await step("search_record", async () => {
    await page.getByRole('button', { name: 'Go' }).click();
    await page.locator('css=button.avatar-button').click();
  })
 
  await step("logout", async () => {
    await page.getByText('Logout').click();
  })
}
```

**Comment by bernardobridge** [comment]
Hi @dyllandry 👋 ,

That's just an example of using `step`. You can organise them how it makes most sense for your tests. The example was actually based off of another user's test script that asked for help here in Discussions.

That being said, I agree your suggestion makes more sense, so I went ahead and changed it. Thanks for the suggestion 👍
